Fix SwipeRating to start at the tapped point + Add jumpValue and onSwipeRating props, tap to rate feature.

postmelee opened the pull request:
### Hi, I found that SwipeRating doesn't work well in some cases.
### And I think I can solve these issues with my improvements: #118 #101 #83 #82 #56
--- 
## These are the issues I found and Solved:
### 1. Swiping
Swipe Rating always starts from the default starting value. The ratingValue is only affected by the gesture.dx. So the initial pressed position is needed to affect the ratingValue in **onPanResponderMove**.
### 2. Tapping
Tapping the panResponder only evaluates **onPanResponderGrant**. So the ratingValue and the ratingImage doesn't changes to the position that user pressed.

## And I added some customize options to props.
These are the props I added:
### 1. jumpValue : number (optional)
**jumpValue** makes the ratingValue possible to jump per the value.
For example, if the **jumpValue={0.5}**, the ratingValue will increase/decrease like 0 => 0.5 => 1.0 => 1.5 => 2.0 => 2.5...
else if the **jumpValue={0.2}**, the ratingValue will increase/decrease like 0 => 0.2 => 0.4 => 0.6 => 0.8 => 1.0...
Of course this can be done with **minValue**

### 2. onSwipeRating : ( rating : number) => void (optional)
**onSwipeRating** is a callback that  is evaluated with **onStartRating** , during swiping event and before **onFinishRating**. (when if **onPanResponderGrant** and **onPanResponderMove** is active)
This makes possible to get current value while swiping.
This can make user to using ratingValue to other component.

### 3 (new edit). onStartRating: ( rating: number ) => void
I add `rating` param to `onStartRating` callback, so that user can get the rating value at **tapping** event

## Here are the examples I captured
### 1. minValue={0.5} jumpValue={0.5} show rating in other component by onSwipeRating

https://user-images.githubusercontent.com/65849802/110312497-7bec0c80-8048-11eb-8f32-c8622b8a1629.mov

As you can see, swipe starts at the point that user pressed (1-1), tapping the star works fine (1-2), ratingValues are jumps per the **jumpValue** (2-1), and my information component is showing the ratingValue in real time (2-2).

### 2. minValue={0.5} show rating in other component by onSwipeRating

https://user-images.githubusercontent.com/65849802/110312707-c9687980-8048-11eb-880e-479e07fe3b75.mov

you can use without the **jumpValue** props (optional)

This is the first time for sending pull request. 😅 So let me know if you have any feedback or message. I'll edit READ.me as soon as you accept my pull request! Thank you!!

sshivananda commented:
@postmelee - I am going to take a look at the changes. In the meanwhile - looks like there are a lot of styling changes to the file : which makes it difficult to narrow down to the actual code that was changed. I will add a style guide to the repo - so that styling is done/ maintained automatically by a linter: in the meanwhile - if you could revert code styling changes from this PR - it would be helpful in reviewing the code changes.

postmelee commented:
> @postmelee - I am going to take a look at the changes. In the meanwhile - looks like there are a lot of styling changes to the file : which makes it difficult to narrow down to the actual code that was changed. I will add a style guide to the repo - so that styling is done/ maintained automatically by a linter: in the meanwhile - if you could revert code styling changes from this PR - it would be helpful in reviewing the code changes.
> 
> 
> 
> 

I'll fix the styles and commit again

postmelee commented:
@sshivananda  Hi, I reverted code styling changes in my new commit(updated) and added a feature at `onStartRating`. I am always open to suggestions so please check and feel free to comment me anytime. :)

postmelee commented:
@Monte9 @sshivananda Is there any problem with my request? I think these are important features.

sshivananda commented on src/SwipeRating.js:
```diff
@@ -103,6 +112,14 @@ export default class SwipeRating extends Component {
     }
   }
 
+  handleLayoutChange() {
```
why is this required?

postmelee replied:
The problem was any swipe events are started from the center of the swipe view. This is because we only changed swipe value depending on the dx value from the center of it. I decided to add the relative distance from the center of the swipe view to solve this problem. (center + relativeX + dx)

![image](https://user-images.githubusercontent.com/65849802/111990216-4241f800-8b56-11eb-8ec1-52886a5fac6c.png)
There is only x0 which is the screen coordinates of the responder grant. This is absolute position of screen, and we need the tap event's relative position of the swipe component view 's center to calculate the value. So this function is called in the swiper view's `onLayout` event.

sshivananda commented:
@postmelee Took some time to test these out. Had one question on the PR - the rest looks awesome.
Please update the README as well and we can get these merged in.

postmelee commented:
@sshivananda READ.me changed, review replyed. Thank you

sshivananda commented:
Released in 7.5.0

sshivananda commented:
Released in https://github.com/Monte9/react-native-ratings/releases/tag/v7.5.0

sshivananda commented:
Awesome work on the PR @postmelee 💯

sshivananda commented:
@postmelee  can you take a look at https://github.com/Monte9/react-native-ratings/blob/feature/linter-fixes/src/SwipeRating.js#L126#L127 ?
need some help getting braces around the operators to get the operator precedence correct and remove the eslint-disable on line 126

postmelee commented:
> @postmelee  can you take a look at https://github.com/Monte9/react-native-ratings/blob/feature/linter-fixes/src/SwipeRating.js#L126#L127 ?
> 
> need some help getting braces around the operators to get the operator precedence correct and remove the eslint-disable on line 126

I think we can change it from `centerX: px % Dimensions.get( "window" ).width + width / 2 ` to `centerX: (px % Dimensions.get( "window" ).width) + (width / 2) `
